hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequest.java

```diff
@@ -179,6 +174,20 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
         throw new OMException("Failed to commit key, as " + dbOpenKey +
             "entry is not found in the OpenKey table", KEY_NOT_FOUND);
       }
+
+      List<OmKeyLocationInfo> locationInfoList = new ArrayList<>();
+      long allocatedSize = 0;
+      long blockStorageSize = ozoneManager.getScmBlockSize() *
+          omKeyInfo.getFactor().getNumber();
+      for (KeyLocation keyLocation : commitKeyArgs.getKeyLocationsList()) {
+        OmKeyLocationInfo locationInfo =
+            OmKeyLocationInfo.getFromProtobuf(keyLocation);
+        if (locationInfo.getLength() > 0) {
+          locationInfoList.add(OmKeyLocationInfo.getFromProtobuf(keyLocation));
+        }
+        allocatedSize += blockStorageSize;
+      }
```

Comment:
Changes of the OMKeyCommitRequest are not required. The original logic has ensure correctedSpace is correct.

`long correctedSpace = omKeyInfo.getDataSize() * factor - allocatedSize;`
When the key length is 0:
`long correctedSpace = 0 * factor - allocatedSize;`

Just change BlockOutputStreamEntryPool, the added UT should also run normally.